JavaScript/chapter01/1.5 - OneAway/rroque98_sol.js

```diff
@@ -0,0 +1,43 @@
+/* There are 3 types of edits that can be made on a string:
+insert character, remove character, replace a character
+Given 2 strings write a function that will check if it is
+1 or 0 edits away*/
+
+const isOneAway = (str1, str2) => {
+  if (str1.length === str2.length) {
+    let errorCount = 0;
+    for (let i = 0; i < str1.length; i++) {
+      if (str1[i] !== str2[i]) {
+        errorCount++;
+      }
+      if (errorCount > 1) {
```

Comment:
you don't need to check this unless you actually modify the error count. so you can use the [guard clause pattern](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) and reduce nesting while you're at it:

```javascript
for (let i = 0; i < str1.length; i++) {
  if (str1[i] === str2[i]) {
    continue;
  }
  errorCount++;
  if (errorCount > 1) {
    return false;
  }
}
```

Reply:
Ok, I see what you mean. Thanks for the feedback